tests/test_rollup.py

```
@@ -1020,7 +1020,206 @@ def test_rebuilding_the_same_hour_twice_does_not_double_it() -> None:
+        "INSERT INTO circuit_reading (timestamp, circuit_id, watts) VALUES (?, 1, 900)",
+        (hour + 3599,),
+    )
+    rebuild_circuit_hourly(conn, hour, hour + 7200, cadence_seconds=60)
+
+    rows = dict(
+        (row[0], (row[1], row[2], row[3]))
+        for row in conn.execute(
+            "SELECT timestamp, watts, sample_count, covered_seconds"
+            " FROM circuit_hourly WHERE circuit_id = 1"
+        )
+    )
```

Ruff (0.16.1)

[warning] Unnecessary generator (rewrite as a dict comprehension)

Rewrite as a dict comprehension

(C402)